src/main/java/com/alkemy/ong/web/controllers/UserController.java

```diff
@@ -27,8 +31,21 @@ public ResponseEntity<List<UserDto>> findAll() {
     return ResponseEntity.ok().body(toListDto(userService.findAll()));
   }
 
+  @GetMapping("/{id}")
+  public ResponseEntity<UserDto> findById(@PathVariable Long id) {
+    Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+    Users foundUser = userService.findById(id);
+    String principal = (String) authentication.getPrincipal();
+
+    if (Objects.equals(foundUser.getEmail(), principal) ||
+        authentication.getAuthorities().contains(new SimpleGrantedAuthority("admin"))) {
+      return ResponseEntity.ok(toDto(foundUser));
```

Comment:
off topic, pero me cambias en authController el singIn por sign in en el nombre del metodo?